internal/service/fw/api_access_codes_resource_test.go

```diff
@@ -0,0 +1,408 @@
+	name := acctest.RandomNameWithPrefix("ac")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		CheckDestroy:             testAccCheckAccessCodesDestroy(context.Background(), &v),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAccessCodesBasicConfig(name),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
+					testAccCheckAccessCodesDisappears(context.Background(), &v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAccessCodesResource_Name(t *testing.T) {
+	resourceName := "bloxone_td_access_code.test_name"
+	var v1, v2 fw.AtcfwAccessCode
+	name1 := acctest.RandomNameWithPrefix("ac")
+	name2 := acctest.RandomNameWithPrefix("ac")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesName(name1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v1),
+					resource.TestCheckResourceAttr(resourceName, "name", name1),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.type", "custom_list"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccAccessCodesName(name2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesDestroy(context.Background(), &v1),
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v2),
+					resource.TestCheckResourceAttr(resourceName, "name", name2),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.type", "custom_list"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccAccessCodesResource_Activation(t *testing.T) {
+	resourceName := "bloxone_td_access_code.test_activation"
+	var v fw.AtcfwAccessCode
+	name := acctest.RandomNameWithPrefix("ac")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccAccessCodesActivation(name, time.Now().UTC().Format(time.RFC3339)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAccessCodesExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.data", "terraform_test"),
+					resource.TestCheckResourceAttr(resourceName, "rules.0.type", "custom_list"),
+					resource.TestCheckResourceAttr(resourceName, "activation", time.Now().UTC().Format(time.RFC3339)),
```

Comment:
isn't there a chance that `time.Now()` in L112 and here might be different. You should use a variable instead,

Reply:
Added a variable instead.